appcues/src/main/java/com/appcues/statemachine/StateMachine.kt

```diff
@@ -1,214 +1,77 @@
 package com.appcues.statemachine
 
-import com.appcues.AppcuesConfig
-import com.appcues.AppcuesCoroutineScope
 import com.appcues.action.ActionProcessor
 import com.appcues.analytics.ExperienceLifecycleTracker
 import com.appcues.data.model.Experience
 import com.appcues.statemachine.Action.EndExperience
-import com.appcues.statemachine.Action.RenderStep
 import com.appcues.statemachine.Action.ReportError
-import com.appcues.statemachine.Action.Reset
 import com.appcues.statemachine.Action.StartExperience
-import com.appcues.statemachine.Action.StartStep
 import com.appcues.statemachine.Error.ExperienceAlreadyActive
-import com.appcues.statemachine.SideEffect.AwaitEffect
-import com.appcues.statemachine.SideEffect.ContinuationEffect
-import com.appcues.statemachine.SideEffect.PresentContainerEffect
-import com.appcues.statemachine.SideEffect.ProcessActions
-import com.appcues.statemachine.SideEffect.ReportErrorEffect
-import com.appcues.statemachine.State.BeginningExperience
-import com.appcues.statemachine.State.BeginningStep
-import com.appcues.statemachine.State.EndingExperience
-import com.appcues.statemachine.State.EndingStep
-import com.appcues.statemachine.State.Idling
-import com.appcues.statemachine.State.RenderingStep
-import com.appcues.statemachine.StepReference.StepOffset
-import com.appcues.statemachine.Transition.EmptyTransition
-import com.appcues.statemachine.Transition.ErrorLoggingTransition
-import com.appcues.statemachine.Transitions.Companion.fromRenderingStepToEndingExperience
-import com.appcues.statemachine.Transitions.Companion.fromRenderingStepToEndingStep
+import com.appcues.statemachine.states.IdlingState
 import com.appcues.util.ResultOf
 import com.appcues.util.ResultOf.Failure
 import com.appcues.util.ResultOf.Success
-import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.SharedFlow
-import kotlinx.coroutines.launch
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
 
 internal class StateMachine(
-    private val appcuesCoroutineScope: AppcuesCoroutineScope,
-    private val config: AppcuesConfig,
     private val actionProcessor: ActionProcessor,
     private val lifecycleTracker: ExperienceLifecycleTracker,
-    initialState: State = Idling
+    initialState: State = IdlingState,
+    private val onEndedExperience: (Experience) -> Unit = {},
 ) {
 
     private val _stateFlow = MutableSharedFlow<State>(1)
     val stateFlow: SharedFlow<State>
         get() = _stateFlow
 
-    private val _errorFlow = MutableSharedFlow<Error>(1)
+    private val _errorFlow = MutableSharedFlow<Error>()
```

Comment:
we can keep an eye for it but ideally we dont need to keep the last error since we are emitting errors BEFORE we stop the experienceLifetracking, so this is so when we start observing again we dont get old value (like we had before where we had to call resetReplayCache()